knowledge-graph/src/main/scala/io/renku/knowledgegraph/Failure.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.knowledgegraph
+
+import io.renku.data.Message
+import org.http4s.{Response, Status}
+
+private sealed trait Failure extends Exception {
```

Comment:
Seems to me a candidate for the not-yet-existing commons library :) (minus the `toResponse` thing)

Reply:
Yeah!